### DISCUSSION TITLE
Dynamic Imports and conditional compilation of external packages

### DISCUSSION THREAD

**Discussion by heyitsarpit** [vercel/next.js#63913] [Help] [answered]
### Summary

next version: v14.1.4

I'm working with a package `@lifi/widget`, it requires extra transpilation to import css files for it's components etc.

To do this they recommend I add `transpilePackages: ['@lifi/widget', '@lifi/wallet-management']` to my `next.config.js` file.

This is acceptable to me for production, but for development it slows down the build time significantly. The number of compiled packages goes from 8088 to 22499.
  
To resolve this I added dynamic loading with dynamic importing with `next/dynamic` and `React.lazy`.
  ```tsx
  const LiFiWidget = dynamic(
  () =>
    ENV.NODE_ENV === 'production'
      ? import('~/ui/Bridge/Lifi').then((m) => m.LiFiWidget)
      : import('~/ui/Bridge/Lifi/Placeholder').then((m) => m.default),
  { ssr: false })
  ```
I expect that `ENV.NODE_ENV === 'production'` will be true only in production, so there will be no need to compile the packages but this is not true and **next.js still tries to compile the packages and fails with an error** because I disabled `transpilePackages` options in dev mode.

TLDR: How do I make next.js not compile the packages I don't need in dev mode, while using dynamic imports.

**Comment by heyitsarpit** [comment, marked as answer]
Solved it!, I wasn't getting good answers from claude sonnet so didn't expected to solve it with AI tools but I tried GPT 3.5 and it solved it immediately 

```js
const webpack = require('webpack')

const nextConfig = {
  transpilePackages:
    process.env.NODE_ENV === 'production'
      ? ['@lifi/widget', '@lifi/wallet-management']
      : undefined,
  webpack: (config, { isServer }) => {
    if (!isServer && process.env.NODE_ENV === 'development') {
      config.plugins.push(
        new webpack.IgnorePlugin({
          resourceRegExp: /^@lifi\/widget$/
        }),
        new webpack.IgnorePlugin({
          resourceRegExp: /^@lifi\/wallet-management$/
        })
      )
    }

    return config
  }
  ```

**Reply by heyitsarpit** [reply]
Initial compile time went from `34.1s (22495 modules) -> 15.7s (8093 modules)`
